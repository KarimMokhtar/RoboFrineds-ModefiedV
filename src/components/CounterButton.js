import React, { useState } from "react";

const CounterButton = React.memo((props) => {
  const [count, setCount] = useState(0);
  return (
      <button
        id="counter"
        className={`clicks-${count}`}
        color={props.color}
        onClick={() => setCount((count) => count + 1)}
      >
        Count: {count}
      </button>
  );
});

export default CounterButton;
