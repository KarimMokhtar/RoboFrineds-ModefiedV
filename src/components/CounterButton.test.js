import React from 'react';

import { shallow, mount } from 'enzyme';
import CounterButton from './CounterButton';

describe('test the counter component', () => {
  expect.assertions(1);
  it('smapshots component', () => {
    const mockColor = 'red';
    const wrapper = shallow(<CounterButton color={mockColor} />);
    expect(wrapper.debug()).toMatchSnapshot();
  });

  expect.assertions(1);
  it('the counter works correctly', () => {
    const mockColor = 'red';
    const wrapper = mount(<CounterButton color={mockColor} />);

    wrapper.find('#counter').simulate('click');
    wrapper.find('#counter').simulate('click');
    expect(wrapper.find('.clicks-2').length).toEqual(1);
    expect(wrapper.find('.clicks-0').length).toEqual(0);

    wrapper.find('#counter').simulate('click');
    expect(wrapper.find('.clicks-3').length).toEqual(1);
    expect(wrapper.props()).toEqual({ color: 'red' });
  });
});
