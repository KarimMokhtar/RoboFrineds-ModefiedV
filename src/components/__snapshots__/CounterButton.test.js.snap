// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test the counter component smapshots component 1`] = `
"<button id=\\"counter\\" className=\\"clicks-0\\" color=\\"red\\" onClick={[Function: onClick]}>
  Count: 
  0
</button>"
`;
